### ISSUE TITLE
IDE0047 false positive in dynamic expression

### ISSUE THREAD

**Issue by chrarnoldus** [dotnet/roslyn#47365]
**Version Used**:

Microsoft Visual Studio Enterprise 2019
Version 16.7.2, 16.8.0 Preview 2.1

**Steps to Reproduce**:

Enter the following code in the code editor:

```c#
using System;

namespace ConsoleApp2
{
    internal class Program
    {
        private static void Main()
        {
            dynamic i = 1;
            dynamic s = "s";
            Console.WriteLine(s + (1 + i)); // <= IDE0047 reported
        }
    }
}
```

**Expected Behavior**:

No suggestion.

**Actual Behavior**:

IDE0047 is reported suggesting removal of the parentheses. This is a false positive, because doing so changes the meaning of the code (`s11` is printed instead of `s2`).

![afbeelding](https://user-images.githubusercontent.com/12196001/91974272-d44ef400-ed1d-11ea-9e7f-24bb5016f2bb.png)

**Comment by Youssef1313** [comment]
I believe the bug is in this part:

https://github.com/dotnet/roslyn/blob/2b12b9f3b8fe8895694280648b523bc4fd95211f/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ParenthesizedExpressionSyntaxExtensions.cs#L13-L75

Probably this method should special-case the "dynamic" case and just return false for that case.
